test/e2e/framework/ingress.go

```diff
@@ -0,0 +1,121 @@
+// Copyright 2016 The prometheus-operator Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package framework
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/yaml"
+	"k8s.io/client-go/pkg/api/v1"
+	"k8s.io/client-go/pkg/apis/extensions/v1beta1"
+	"k8s.io/client-go/pkg/util/intstr"
+	"os"
+	"time"
+)
+
+func (f *Framework) MakeBasicIngress(serviceName string, servicePort int) *v1beta1.Ingress {
+	return &v1beta1.Ingress{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "monitoring",
+		},
+		Spec: v1beta1.IngressSpec{
+			Rules: []v1beta1.IngressRule{
+				v1beta1.IngressRule{
+					IngressRuleValue: v1beta1.IngressRuleValue{
+						HTTP: &v1beta1.HTTPIngressRuleValue{
+							Paths: []v1beta1.HTTPIngressPath{
+								v1beta1.HTTPIngressPath{
+									Backend: v1beta1.IngressBackend{
+										ServiceName: serviceName,
+										ServicePort: intstr.FromInt(servicePort),
+									},
+									Path: "/metrics",
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func (f *Framework) CreateIngress(i *v1beta1.Ingress) error {
+	_, err := f.KubeClient.Extensions().Ingresses(f.Namespace.Name).Create(i)
+	return err
+}
+
+func (f *Framework) SetupNginxIngressControllerIncDefaultBackend() error {
+	// Create Nginx Ingress Replication Controller
+	if err := createReplicationControllerViaYml("./framework/ressources/nxginx-ingress-controller.yml", f); err != nil {
```

Comment:
I had some thoughts on this but left them out before.
I advocated for not using plain YAML files before in other places – e.g. in cases where operators create objects. But as we see in the case of the ingress that gets really noisy and nearly impossible to read.

Should we use YAMLs everywhere then? What about those where we insert dynamic values. We could use `text/template` there to populate them.